flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusPushGatewayReporter.java

```diff
@@ -57,6 +60,17 @@
         this.deleteOnShutdown = deleteOnShutdown;
     }
 
+    PrometheusPushGatewayReporter(
+            String hostUrl,
+            String jobName,
```

Comment:
Why do we need a second constructor and can't just change the existing one?

Reply:
Ah, I misinterpreted the public evolving annotation. Thought I shouldn't introduce breaking changes to this class, will clean it up.